beacon-chain/rpc/eth/shared/structs_blocks_conversions.go

```diff
@@ -0,0 +1,3099 @@
+	if src == nil {
+		return nil, errors.New("VoluntaryExits are empty, nothing to convert.")
+	}
+	exits := make([]*SignedVoluntaryExit, len(src))
+	for i, e := range src {
+		exits[i] = &SignedVoluntaryExit{
+			Message: &VoluntaryExit{
+				Epoch:          fmt.Sprintf("%d", e.Exit.Epoch),
+				ValidatorIndex: fmt.Sprintf("%d", e.Exit.ValidatorIndex),
+			},
+			Signature: hexutil.Encode(e.Signature),
+		}
+	}
+	return exits, nil
+}
+
+func ConvertBlsChanges(src []*SignedBlsToExecutionChange) ([]*eth.SignedBLSToExecutionChange, error) {
+	if src == nil {
+		return nil, errors.New("nil b.Message.Body.BlsToExecutionChanges")
+	}
+	err := VerifyMaxLength("b.Message.Body.BlsToExecutionChanges", len(src), 16)
+	if err != nil {
+		return nil, err
+	}
+	changes := make([]*eth.SignedBLSToExecutionChange, len(src))
+	for i, ch := range src {
+		sig, err := DecodeWithLength(ch.Signature, fieldparams.BLSSignatureLength)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not decode b.Message.Body.BlsToExecutionChanges[%d].Signature", i)
+		}
+		index, err := strconv.ParseUint(ch.Message.ValidatorIndex, 10, 64)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not decode b.Message.Body.BlsToExecutionChanges[%d].Message.ValidatorIndex", i)
+		}
+		pubkey, err := DecodeWithLength(ch.Message.FromBlsPubkey, fieldparams.BLSPubkeyLength)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not decode b.Message.Body.BlsToExecutionChanges[%d].Message.FromBlsPubkey", i)
+		}
+		address, err := DecodeWithLength(ch.Message.ToExecutionAddress, common.AddressLength)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not decode b.Message.Body.BlsToExecutionChanges[%d].Message.ToExecutionAddress", i)
+		}
+		changes[i] = &eth.SignedBLSToExecutionChange{
+			Message: &eth.BLSToExecutionChange{
+				ValidatorIndex:     primitives.ValidatorIndex(index),
+				FromBlsPubkey:      pubkey,
+				ToExecutionAddress: address,
+			},
+			Signature: sig,
+		}
+	}
+	return changes, nil
+}
+
+func ConvertInternalBlsChanges(src []*eth.SignedBLSToExecutionChange) ([]*SignedBlsToExecutionChange, error) {
+	if src == nil {
+		return nil, errors.New("BlsToExecutionChanges are empty, nothing to convert.")
+	}
+	changes := make([]*SignedBlsToExecutionChange, len(src))
+	for i, ch := range src {
+		changes[i] = &SignedBlsToExecutionChange{
+			Message: &BlsToExecutionChange{
+				ValidatorIndex:     fmt.Sprintf("%d", ch.Message.ValidatorIndex),
+				FromBlsPubkey:      hexutil.Encode(ch.Message.FromBlsPubkey),
+				ToExecutionAddress: hexutil.Encode(ch.Message.ToExecutionAddress),
+			},
+			Signature: hexutil.Encode(ch.Signature),
+		}
+	}
+	return changes, nil
+}
+
+func uint256ToSSZBytes(num string) ([]byte, error) {
+	uint256, ok := new(big.Int).SetString(num, 10)
+	if !ok {
+		return nil, errors.New("could not parse Uint256")
+	}
+	if !math.IsValidUint256(uint256) {
+		return nil, errors.New("is")
+	}
```

Comment:
Error message is incomplete. But do we even have to validate this once `ok` passes?

Reply:
oh weird ok not sure how that cut off will fix